source/kernel/src/drivers/serial_mux.rs

```diff
@@ -5,33 +5,31 @@ use crate::{
         kchannel::{KChannel, KConsumer},
         oneshot::Reusable,
     },
-    registry::{
-        simple_serial::SimpleSerial, Envelope, KernelHandle, Message, RegisteredDriver, ReplyTo,
-    },
+    drivers::simple_serial::SimpleSerialClient,
+    registry::{Envelope, KernelHandle, Message, RegisteredDriver, ReplyTo},
     Kernel,
 };
 use maitake::sync::Mutex;
 use mnemos_alloc::containers::{HeapArc, HeapArray, HeapFixedVec};
 use uuid::Uuid;
 
+////////////////////////////////////////////////////////////////////////////////
+// Service Definition
+////////////////////////////////////////////////////////////////////////////////
+
 /// SerialMux is the registered driver type
```

Comment:
maybe worth a comment explaining that this module implements the serial mux server as well as the client and registered driver?